### ISSUE TITLE
missing instruction for BahdanauAttention

### ISSUE THREAD

**Issue by ad26kt** [tensorflow/tensorflow#15199]
This is just my opinion: 
when call BahdanauAttention instance , it create a variable scope with None name_or_scope variable.
While name_or_scope is None, and get_variable with the same name inside the variable scope repeatedly, it will automatically add '_N' to the name of the scope. And I think it is not compatible with some functions like stati_rnn, because there are explicit 'for loop' inside the function and every loop of 'for loop' will create different variables inside the variable scope but not creating once and sharing

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
Have I written custom code
OS Platform and Distribution
TensorFlow installed from
TensorFlow version
Bazel version
CUDA/cuDNN version
GPU model and memory
Exact command to reproduce

**Comment by ad26kt** [comment]
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: Yes, but only with a few lines.
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Ubuntu 16.04
- **TensorFlow installed from (source or binary)**: conda pip install with the instruction on the tensorflow website(binary)
- **TensorFlow version (use command below)**: 1.4
- **Python version**: 2.7
- **Bazel version (if compiling from source)**:
- **GCC/Compiler version (if compiling from source)**:
- **CUDA/cuDNN version**: 8.0
- **GPU model and memory**: gtx 1080 ti
- **Exact command to reproduce**:
1. I created an GRUCell
2. created an Bahdanau attention
3. add wrapper to GRUCell with attention
4. run static_rnn

I think it doesn't work because 1. there is a for loop inside ststic_rnn  2. there is a variable_scope inside bahdanau attention(or luong attention) whose first argument(name_or_scope) is given None variable   ->> while name_or_scope is None, each time a variable within the variable scope is created, the name of variable scope will be added '_N'

**Comment by ad26kt** [comment]
I updated the Exact command to reproduce field

**Comment by angersson** [comment]
I'm having a hard time understanding what the issue here is. Can you rephrase your question with an explicit bug report or feature request?

**Comment by ad26kt** [comment]
I uploaded an exact code to reproduce the error up there.
In a nutshell, an rnn cell with attention wrapper can only be used with decoder function but not rnn function

**Comment by angersson** [comment]
@lukaszkaiser, can you take a look at this?

**Comment by lukaszkaiser** [comment]
I believe you're right in everything you say. The only issue is that `static_rnn` is deprecated. It'd still be good to adjust BahdanuAttention to be better for sharing (e.g., take a name parameter like all layers). Feel free to send a PR if you have some code, of course! In other case, I don't think this is a high-priority problem as it seems to only affect deprecated code.

**Comment by ad26kt** [comment]
Actually I simply changed the `with variable_scope.variable_scope(None, "bahdanau_attention", [query]):` to `with variable_scope.variable_scope("bahdanau_attention"):` and it works fine for static_rnn. And after all, I changed to the combination of attention + dynamic_rnn and it works fine.
